src/stream/src/executor/managed_state/join/mod.rs

```diff
@@ -306,7 +350,7 @@ impl<K: HashKey, S: StateStore> JoinHashMap<K, S> {
     /// Insert a key
     pub fn insert(&mut self, join_key: &K, pk: Row, value: JoinRow) -> StreamExecutorResult<()> {
         if let Some(entry) = self.inner.get_mut(join_key) {
-            entry.insert(pk, value.clone());
+            entry.insert(pk, value.encode()?);
```

Comment:
Anw, maybe the “clean” way to unify the state_table encoding interface and the row encode interface is to offload all the encoding logic into state_table, and offer state_table insertion from encoded_row, so that we save on encoding twice.

Reply:
Alternately, state_table.insert can return a reference to the encoded row, in which case inserting into the cache is as simple as copying the relevant bytes.

Reply:
Yes! We are discussing the new interface of state table. We expect to finalize the design soon.

Reply:
Both +1. We should really rethink the current design of the state table.

Reply:
It may also be worth pointing out that delete + insert (i.e. update) encodes the PK twice.